daemon/api/endpoints/logs.py

```diff
@@ -27,64 +34,88 @@ async def _export_logs(workspace_id: uuid.UUID, log_id: uuid.UUID):
         return FileResponse(filepath)
 
 
-class LogStreamingEndpoint(WebSocketEndpoint):
-    def __init__(self, scope: Scope, receive: Receive, send: Send) -> None:
-        super().__init__(scope, receive, send)
-        # Accessing path / query params from scope in ASGI
-        # https://asgi.readthedocs.io/en/latest/specs/www.html#websocket-connection-scope
-        info = self.scope.get('path').split('/')
-        workspace_id = info[-2]
-        log_id = info[-1]
-        self.filepath = get_workspace_path(workspace_id, log_id, 'logging.log')
-        self.active_clients = []
+def _websocket_details(websocket: WebSocket):
+    return f'{websocket.client.host}:{websocket.client.port}'
 
-    async def on_connect(self, websocket: WebSocket) -> None:
-        await websocket.accept()
 
-        self.client_details = f'{websocket.client.host}:{websocket.client.port}'
-        self.active_clients.append(websocket)
-        daemon_logger.info(f'{self.client_details} is connected to stream logs!')
+# TODO for now contian a single connection. Ideally there must be one
+# manager per log with a thread checking for updates in log and broadcasting
+# to active connections
+class ConnectionManager:
```

Comment:
It may be good to have this as a `context` manager to have a more robust connection management?

Reply:
@JoanFM [This is](https://fastapi.tiangolo.com/advanced/websockets/#handling-disconnections-and-multiple-clients) how fastapi recommends the websockets implementation.

Reply:
Ok